### PR TITLE
[Filebeat] Add network direction processor to Zeek and Suricata modules

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds the `add_network_direction` processor to the Zeek & Suricata module filesets using the `internal_networks` variable. The `internal_networks`  variable is set to default to `[ private ]` and the `add_network_direction` will only run if that varaible is defined.

## Why is it important?

The `add_network_direction` process adds the `network.direction` to documents which allows users to easily filter on traffic based on direction without needing to know specific IP subnets.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

```
cd beats/x-pack/filebeat
GENERATE=true TESTING_FILEBEAT_MODULES=suricata,zeek mage -v pythonIntegTest
```
## Related issues

N/A

### CODE REVIEW

**Comment by legoguy1000** [comment]
@andrewkroh I think I'm almost done with this PR but I think I found a bug and want to check with you first.  I noticed on the Snort and Sonicwall JS scripts, for source and dest IP it was using the `fld_append` function.  This causes the field to be an array/list even if its a single value.  When I tried to add the `add_network_direction` processor, it didn't work until I change the JS function to `fld_set` so its just a single value. My thought is that if it is an array/list, the processor should just use the first value.  Thoughts??

**Comment by P1llus** [comment]
After some discussion, we decided that all of the network_direction parts would have to stay in edge processing, to keep the module backwards compatible.
The new version that has edge processing removed will be in the package versions.

**Comment by P1llus** [comment]
run tests

**Comment by P1llus** [comment]
run tests

**Comment by P1llus** [comment]
run tests

**Comment by legoguy1000** [comment]
changed back to draft as I realized I never added the update to the docs for the new `internal_networks` variable.  Will do that then we should be good to merge.

From Netflow module.
```
var.internal_networks
A list of CIDR ranges describing the IP addresses that you consider internal. This is used in determining the values of source.locality, destination.locality, and flow.locality. The values can be either a CIDR value or one of the named ranges supported by the network condition. The default value is [private] which classifies RFC 1918 (IPv4) and RFC 4193 (IPv6) addresses as internal.
```

**Comment by legoguy1000** [comment]
Docs updated

**Comment by P1llus** [comment]
run tests

**Comment by P1llus** [comment]
run tests

File: x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js [hunk 1]
```diff
@@ -1007,8 +1007,8 @@ var ecs_mappings = {
     "child_process": {to:[{field: "process.name", setter: fld_prio, prio: 1}]},
     "city.dst": {to:[{field: "destination.geo.city_name", setter: fld_set}]},
     "city.src": {to:[{field: "source.geo.city_name", setter: fld_set}]},
-    "daddr": {convert: to_ip, to:[{field: "destination.ip", setter: fld_append},{field: "related.ip", setter: fld_append}]},
-    "daddr_v6": {convert: to_ip, to:[{field: "destination.ip", setter: fld_append},{field: "related.ip", setter: fld_append}]},
+    "daddr": {convert: to_ip, to:[{field: "destination.ip", setter: fld_set},{field: "related.ip", setter: fld_append}]},
```
Comment (andrewkroh) [inline x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js:1010]: Why set and not append? Normally related.ip is an array. This seems unrelated to adding network direction.

**Review by andrewkroh — COMMENTED** [review]
Modifying snort and sonicwall will cause issues in the future since those are generated modules from https://github.com/adriansr/nwdevice2filebeat. This feature could be added to the generator then all of the RSA would get this processor. IMO I think we should remove the changes from those two modules and proceed with this PR.

File: x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js (hunk 1 above)
Comment (legoguy1000) [reply x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js:1010]: i don't recall removing `"daddr_v6": {convert: to_ip, to:[{field: "destination.ip", setter: fld_append},{field: "related.ip", setter: fld_append}]},`.  That is probably an accident.  As for the change from `{field: "destination.ip", setter: fld_append}` to `{field: "destination.ip", setter: fld_set}`, is because the `network_direction` processor requires a string field not an array.

Comment (legoguy1000) [reply x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js:1010]: The append for `related.ip` is unchanged.

Comment (legoguy1000) [reply x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js:1010]: I am reverting the changes for Snort and Sonicwall.

Comment (legoguy1000) [reply x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js]: removed

Comment (andrewkroh) [reply x-pack/filebeat/module/sonicwall/firewall/config/liblogparser.js]: Thanks for pointing out at source.ip and destination.ip should be scalars. I think we should make that change just not in this PR. Would you please open an issue for this? It does look like you reverted most of the snort/sonicwall changes, but a small part of snort is still changed.

**Review by andrewkroh — APPROVED** [review]
